### PR TITLE
Call execute operations with list

### PR DESCRIPTION
**Pull request by PatrykGala**
## Before submitting checklist

- [x] Did you **update the CHANGELOG**? (not for test updates, internal changes/refactors or CI/CD setup)
- [x] Did you **ask the docs owner** to review all the user-facing changes?

### CODE REVIEW

File: src/neptune/internal/backends/hosted_neptune_backend.py [hunk 1]
```diff
@@ -638,13 +637,17 @@ def _execute_operations(
...
         }
 
-        try:
-            result = self.leaderboard_client.api.executeOperations(**kwargs).response().result
-            return [MetadataInconsistency(err.errorDescription) for err in result]
-        except HTTPNotFound as e:
-            raise ContainerUUIDNotFound(container_id, container_type) from e
-        except (HTTPPaymentRequired, HTTPUnprocessableEntity) as e:
-            raise NeptuneLimitExceedException(reason=e.response.json().get("title", "Unknown reason")) from e
+        @with_api_exceptions_handler
```
Comment (PatrykGala) [inline src/neptune/internal/backends/hosted_neptune_backend.py:640]:
```
        @with_api_exceptions_handler
        def _execute():
            result = self.leaderboard_client.api.executeOperations(**kwargs).response().result
            return [MetadataInconsistency(err.errorDescription) for err in result]

        try:
            return _execute()
        except HTTPNotFound as e:
            raise ContainerUUIDNotFound(container_id, container_type) from e
        except (HTTPPaymentRequired, HTTPUnprocessableEntity) as e:
            raise NeptuneLimitExceedException(reason=e.response.json().get("title", "Unknown reason")) from e
```
?

File: src/neptune/internal/backends/neptune_backend.py [hunk 2]
```diff
@@ -300,8 +299,8 @@ def fetch_atom_attribute_values(
     def search_leaderboard_entries(
         self,
         project_id: UniqueId,
-        types: Optional[Iterable[ContainerType]] = None,
+        types: Optional[List[ContainerType]] = None,
```
Comment (Raalsky) [inline src/neptune/internal/backends/neptune_backend.py:302]: Why do we have to update typings here?

Comment (PatrykGala) [reply src/neptune/internal/backends/neptune_backend.py:302]: to avoid a `Iterable` in the future

Comment (Raalsky) [reply src/neptune/internal/backends/neptune_backend.py:302]: We're not chaining here so what will go wrong here?

**Review by Raalsky — APPROVED** [review]

**Pull request merged**